keylime-agent/src/crypto.rs

```diff
@@ -75,6 +89,66 @@ pub(crate) fn write_x509(cert: &X509, file_path: &Path) -> Result<()> {
     Ok(())
 }
 
+/// Check an x509 certificate contains a specific public key
+pub(crate) fn check_x509_key(
+    cert: &X509,
+    tpm_key: tss_esapi::structures::Public,
+) -> Result<bool> {
+    // Id:RSA_PSS only added in rust-openssl from v0.10.59; remove this let and use Id::RSA_PSS after update
+    // Id taken from https://boringssl.googlesource.com/boringssl/+/refs/heads/master/include/openssl/nid.h#4039
+    let id_rsa_pss: Id = Id::from_raw(912);
+    match cert
+        .public_key()
+        .unwrap() //#[allow_ci]
```

Comment:
`unwrap()` should never be used as it can lead to a panic which should never happen in any circumstance. We can tolerate it in tests, but not in the actual code.

Sorry for being strict, I know there are still some `unwrap()` lingering around, but let's do our best to not add new `unwrap()` usage.